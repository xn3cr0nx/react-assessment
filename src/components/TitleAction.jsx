import React from "react";
import PropTypes from "prop-types";
import Button from "components/Button.jsx";

import "css/titleAction.css"

const TitleAction = props => {
	return (
		<div className="title">
			<h2>{props.title}</h2>
			<Button onClick={props.action}>{props.button}</Button>
		</div>
	);
};

TitleAction.propTypes = {
	title: PropTypes.string,
	button: PropTypes.string,
	action: PropTypes.func
}

export default TitleAction;
