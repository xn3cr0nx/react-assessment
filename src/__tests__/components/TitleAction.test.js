import React from "react";
import { shallow } from "enzyme";
import TitleAction from "components/TitleAction.jsx";

const props = {
	title: "test",
	button: "button_test",
	action: jest.fn()
};

it("renders TitleAction", () => {
	const wrapper = shallow(<TitleAction {...props} />);

	expect(wrapper.find("div").hasClass("title")).toBe(true);
	expect(wrapper.find("h2").text()).toBe("test");

	expect(wrapper).toMatchSnapshot();
});
