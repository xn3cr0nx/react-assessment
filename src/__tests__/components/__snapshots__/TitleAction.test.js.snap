// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TitleAction 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TitleAction
    action={[MockFunction]}
    button="button_test"
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          test
        </h2>,
        <ForwardRef
          onClick={[MockFunction]}
        >
          button_test
        </ForwardRef>,
      ],
      "className": "title",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "button_test",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "button_test",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
	width: 4vw;
	margin: 0.2rem;
	padding: 0.5rem 1rem;
	border: 0.3rem double black;
	border-radius: 0.5rem;
	cursor: pointer;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            test
          </h2>,
          <ForwardRef
            onClick={[MockFunction]}
          >
            button_test
          </ForwardRef>,
        ],
        "className": "title",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "button_test",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "button_test",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
	width: 4vw;
	margin: 0.2rem;
	padding: 0.5rem 1rem;
	border: 0.3rem double black;
	border-radius: 0.5rem;
	cursor: pointer;
",
              ],
            },
            "displayName": "styled.button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
